app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -108,31 +240,34 @@ constructor(
   }
 
   fun onTabChanged(tab: Tab, state: MutableStateFlow<EventsOverviewUiState> = _uiState) {
-    state.value = state.value.copy(tab = tab)
+    // Reset search and filter when tab is changed
+    state.value =
+        state.value.copy(
+            searchQuery = "",
+            isSearchActive = false,
+            isFilterDialogOpen = false,
+            isFilterActive = false,
+            radiusQuery = "",
+            isFreeSwitchOn = false,
+            categoriesCheckedList = mutableSetOf(),
+            tab = tab)
   }
 
   fun onViewListStatusChanged(state: MutableStateFlow<EventsOverviewUiState> = _uiState) {
     state.value = state.value.copy(viewList = !state.value.viewList)
   }
-
-  fun changeFilterDialogOpen() {
-    val current = _uiState.value.isFilterDialogOpen
-    _uiState.update { currentState -> currentState.copy(isFilterDialogOpen = !current) }
-  }
-
-  fun onSearchQueryChange(newQuery: String) {
-    _uiState.update { currentState -> currentState.copy(searchQuery = newQuery) }
-  }
 }
 
 data class EventsOverviewUiState(
     val eventList: EventList = EventList(emptyList(), emptyList(), null),
     val upcomingEventList: EventList = EventList(emptyList(), emptyList(), null),
     val searchQuery: String = "",
+    val isSearchActive: Boolean = false,
     val isFilterDialogOpen: Boolean = false,
-    var radiusInputFilter: Double = -1.0,
-    var freeEventsFilter: Boolean = false,
-    val categorySelectionFilter: List<EventCategory> = emptyList(),
+    val isFilterActive: Boolean = false,
+    val radiusQuery: String = "",
+    val isFreeSwitchOn: Boolean = false,
+    val categoriesCheckedList: MutableSet<EventCategory> = mutableSetOf(),
```

Comment:
In MVVM architecture, it’s typically better to keep the UI state as immutable as possible to prevent unwanted side effects. Consider changing this val to `Set<EventCategory>`.

Reply:
Since this Set is in the ViewModel, it can be mutable. It only needs to be immutable in the UI. i.e. the view should not change the state.

Reply:
See prior comment.